docs/memory-server.md

```diff
@@ -0,0 +1,117 @@
+# Memory server
+
+While metrictank persists timeseries data to cassandra, it can - and should - serve most data out of memory.
+It has two mechanisms to support this: the ring buffers, and the chunk-cache.  These features are complementary to each other, and are described in more detail below.
+
+## In-memory components
+
+### Ring buffers
+
+The ring buffer is simply a list of chunks - one for each series - that holds the latest data for each series that has been ingested (or generated, for rollup series).
+You can configure how many chunks to retain (`numchunks`).
+* The main function of the ring buffers is to keep secondaries able to satisfy queries from RAM, even if the primary is not able to save its chunks instantly, or if the primary
+ crashed and needs to be restarted.  Effectively, the more data in your ring buffer, the longer outages of a primary you can sustain. (up to `(numchunks-1) * chunkspan` in duration)
+* For keeping a "hot cache" of frequently accessed data, this is not necessarily an effective solution, since the same `numchunks` is applied to all raw series
+(and aggregation settings are applied to all series in the same fashion, so a given rollup frequency will have the same `numchunks` for all series)
+So unless you're confident your metrics are all subject to queries of the same timeranges, and that they are predictable, you should look at the chunk cache below.
+
+Note:
+* the last (current) chunk is always a "work in progress", so depending on what time it is, it may be anywhere between empty and full.
+* when metrictank starts up, it will not refill the ring buffer with data from Cassandra. They only fill based on data that comes in.  But once data has been seen, the buffer
```

Comment:
`metrictank` is a name so I think it should be upper case.
there are two spaces before the `But`.